AddUserScreen.tsx

```diff
@@ -1,91 +1,207 @@
+                  autoCapitalize='none'
+                  onEndEditing={(event) => this.verifyCPF(event.nativeEvent.text)}
+                />
+                <Text style={{ fontSize: 12, marginBottom: 25, fontStyle: 'italic' }}>(Utilize pontos e traços)</Text>
+
+                <Text style={{ fontSize: 15, marginBottom: 5 }}>Data de Nascimento:</Text>
+                <TextInput
+                  style={{ height: 40, width: 200, borderColor: 'gray', borderWidth: 1, marginBottom: 0 }}
+                  autoCapitalize='none'
+                  onEndEditing={(event) => this.verifyBirthdate(event.nativeEvent.text)}
+                />
+                <Text style={{ fontSize: 12, marginBottom: 25, fontStyle: 'italic' }}>(Formato: dd/mm/yyyy)</Text>
+
+                <Text style={{ fontSize: 15, marginBottom: 5 }}>E-mail:</Text>
+                <TextInput
+                  style={{ height: 40, width: 200, borderColor: 'gray', borderWidth: 1, marginBottom: 25 }}
+                  autoCapitalize='none'
+                  onEndEditing={(event) => this.verifyEmail(event.nativeEvent.text)}
+                />
+
+                <Text style={{ fontSize: 15, marginBottom: 5 }}>Função:</Text>
+                <Picker
+                  selectedValue={this.state.userRole}
+                  onValueChange={(itemValue) => {
+                    this.verifyRole(itemValue);
+                    this.setState({ userRole: itemValue });
+                  }}>
+                  <Picker.Item label="" value="empty" />
+                  <Picker.Item label="System Administrator" value="admin" />
+                  <Picker.Item label="Regular User" value="user" />
+                </Picker>
+
+                <Button
+                  title="Cadastrar"
+                  color="#9400D3"
+                  onPress={() => {
+                    this.handleButtonPress(mutateFunction);
+                  }}
+                  disabled={this.disabledButton() || loading}
+                />
+
+                <View style={{ alignItems: 'center' }}>
+                  <Text style={{ fontSize: 15, color: "#FF0000", marginTop: 25 }}>{this.state.errorMessage}</Text>
+                </View>
+
+                {loading &&
+                  <ActivityIndicator
+                    size="large"
+                    color="#0000ff"
+                    style={{
+                        zIndex: 0,
+                        position: 'absolute'
+                    }} />
+                }
+              </View>
+            );
+          }
+          }
+        </Mutation>
+      </ApolloProvider>
     );
   }
 
-  private handleButtonPress = (): void => {
+  private async handleButtonPress(mutateFunction): Promise<boolean>  {
+    try {
+      const result = await mutateFunction({ variables: {
+        name: this.state.name, email: this.state.email, cpf: this.state.cpf,
+        birthDate: this.birthdateYYYYMMDD, password: INITIAL_PASSWORD, role: this.state.userRole
```

Comment:
Será que precisa dessa varriável `this.birthdateYYYYMMDD` guardada na classe? Não fica melhor você fazer a conversão a partir do `this.state.birthDate` quando chegar aqui, logo antes de enviar a mutation? O que acha?

Reply:
Sim, mudei o tipo da variável `birthDate`no `state` pra `string`.
Antes estava armazenando um `Date`, mas não era prático.

Reply:
Na vdd, quis dizer que não precisava da variável.